src/hats.ts

```diff
@@ -380,3 +415,58 @@ function removeHat(hat: Hat, event: TransferSingle): void {
   hatBurnedEvent.hat = hatIdToHex(event.params.id);
   hatBurnedEvent.save();
 }
+
+function createDummyTree(treeDomain: string): void {
+  let tree = new Tree(treeDomain);
+  tree.save();
+}
+
+function createDummyHats(hatId: string, contractAddress: Address): void {
+  for (let i = 10; i < hatId.length; i += 4) {
+    let currentHatId = hatId.substring(0, i).padEnd(66, "0");
+    let hat = Hat.load(currentHatId);
+    if (hat === null) {
```

Comment:
do we need this `hat === null` check both here and on line 83?

Reply:
In the case of linking to a hat, we iterate over the array of admins of this hat until the hat itself, starting from the top hat. For each one, we check if it exists by loading it and comparing to null, and if not, create a dummy.
The iteration is from the top hat, because when creating a hat we need the admin to exist in order to update that property in the hat.

The same also in line 83. When a hat is created, we don't know if its admin exists. If not, we'll create all of its missing admins